src/scripts/main.js

```diff
@@ -1,3 +1,285 @@
+function addSeparator(num) {
+  return num.toLocaleString('en-US');
+}
+
+function CustomNotification(title, description, type) {
+  this.title = title;
+  this.description = description;
+  this.type = type;
+}
+
+const pushNotification = ({ title, description, type }) => {
+  const notification = document.createElement('div');
+
+  notification.classList.add('notification', `${type}`);
+  notification.style.top = 15 + 'px';
+  notification.style.right = 15 + 'px';
+
+  const fragmentNot = document.createDocumentFragment();
+  const titleNot = document.createElement('h2');
+
+  titleNot.classList.add('title');
+  titleNot.textContent = title;
+
+  const descriptionNot = document.createElement('p');
+
+  descriptionNot.textContent = description;
+
+  fragmentNot.append(titleNot);
+  fragmentNot.append(descriptionNot);
+  notification.append(fragmentNot);
+
+  document.body.append(notification);
+
+  setTimeout(() => {
+    notification.hidden = true;
+  }, 2000);
+};
+
+formBtn.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const formValues = {
+    employeeName: form.elements.name.value,
+    position: form.elements.position.value,
+    office: form.elements.office.value,
+    age: form.elements.age.value,
+    salary: form.elements.salary.value.trim(),
+  };
+
+  const formSalaryValue =
+    !formValues.salary || isNaN(formValues.salary)
+      ? ''
+      : '$' + addSeparator(+formValues.salary);
+
+  const fields = validateFormFields(formValues);
+
+  pushNotification(fields[0]);
+
+  if (fields[0].type === 'success') {
+    const row = addRow([
+      formValues.employeeName,
+      formValues.position,
+      formValues.office,
+      formValues.age,
+      formSalaryValue,
+    ]);
+
+    tbody.append(row);
+  }
+});
+
+function validateFormFields({ employeeName, position, age, salary }) {
+  const notifications = [];
+
+  if (employeeName.length < 4) {
+    notifications.push(
+      new CustomNotification(
+        'Error',
+        'Name must be at least 4 characters long',
+        'error',
+      ),
+    );
+  }
+
+  if (!position.trim()) {
+    notifications.push(
+      new CustomNotification('Error', 'Position cannot be empty', 'error'),
+    );
+  }
+
+  if (age < 18 || age > 90) {
+    notifications.push(
+      new CustomNotification('Error', 'Age must be between 18 and 90', 'error'),
+    );
+  }
+
+  if (salary <= 0) {
+    notifications.push(
+      new CustomNotification(
+        'Error',
+        'The value of salary must be a number greater than 0',
+        'error',
+      ),
+    );
+  }
+
+  if (notifications.length === 0) {
+    notifications.push(
+      new CustomNotification('Success', 'Data successfully added', 'success'),
+    );
+  }
+
+  return notifications;
+}
+
+function addRow(fields) {
+  const row = document.createElement('tr');
+
+  fields.forEach((field) => {
+    const td = document.createElement('td');
+
+    td.textContent = field;
+    row.append(td);
+  });
+
+  tbody.append(row);
```

Comment:
The `addRow` function appends the row to `tbody` twice: once here and once when the function returns the row. This is unnecessary and could lead to duplicate rows being added. Consider removing this line.